webservices/server-integration/src/main/java/org/jboss/as/webservices/util/SubjectUtil.java

```diff
@@ -174,6 +173,10 @@ public static SecurityIdentity convertToSecurityIdentity(Subject subject, Princi
             }
         }
         identity.withPrivateCredentials(privateCredentials);
+        if (!roles.isEmpty()) {
```

Comment:
Was the change in ordering also required?  Calling withPrivateCredentials should not loose the previously defined roles mapper - if that method does loose them there is a bug in Elytron to address as well.

Reply:
Change ordering is not a required thing and calling withPrivateCredentials doesn't loose mapper.  I just don't want to fix this like: 
```
identity = identity.withRoleMapper(roleCategory, (rolesToMap) -> Roles.fromSet(roles));
```
The root cause of this issue  is identity.WithRoleMapper() won't add the roleMapper in the current identity and only the new created identity which this method returns has this role mapper.